### PR TITLE
Rearrange NetworkServer Broadcast

### PR DESCRIPTION
**Pull request by ThatDan123**
-Old mirror code was loop through all connections and then identities which is expensive

-Because all players currently observe everything we can just loop through all identities and then all connections

-Made some enums bytes
	        
Needs testing

### CODE REVIEW

**Comment by Bod9001** [comment]
only issue is when client is loading They are not observing all the scenes?

**Comment by ThatDan123** [comment]
Ah yeah, that will be an issue

**Comment by ThatDan123** [comment]
@Bod9001 i added a check for observers do you think this will be good enough?

File: UnityProject/Assets/Mirror/Runtime/NetworkServer.cs [hunk 1]
```diff
@@ -1541,6 +1538,8 @@ static void BroadcastToConnections()
 		            //Connection disconnected
 		            if (connection.isReady == false) continue;
 
+		            if (identity.observers.ContainsKey(connection.connectionId) == false) continue;
```
Comment (Bod9001) [inline UnityProject/Assets/Mirror/Runtime/NetworkServer.cs:1541]: I think you could just shortcut this with a bool that gets Set once the client has loaded, tho When we do network bubbling We will require this

**Comment by Bod9001** [comment]
Actually noting about the loop performance, imo it's probably better to have one list of Network behaviours so, the list optimisation I did only removes from one list Versus list for everything a player That is immensely slow

File: UnityProject/Assets/Mirror/Runtime/NetworkServer.cs [hunk 2]
```diff
@@ -1520,63 +1520,40 @@ static void ClearSpawnedDirtyBits()
...
+
+		            //Connection disconnected
+		            if (connection.isReady == false) continue;
+
+		            if (identity.observers.ContainsKey(connection.connectionId) == false) continue;
+
+		            ////////////////////////////
+		            // get serialization for this entity viewed by this connection
+		            // (if anything was serialized this time)
+		            NetworkWriter serialization = GetEntitySerializationForConnection(identity, connection);
```
Comment (ThatDan123) [inline UnityProject/Assets/Mirror/Runtime/NetworkServer.cs:1546]: I'm wondering if a can cache this serialization and reuse it for all the connections, as this is what is mainly draining performance too

**Pull request closed without merging**